Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionActionCocoa.mm

```diff
@@ -417,6 +428,16 @@ - (void)invalidateIntrinsicContentSize
     propertiesDidChange();
 }
 
+void WebExtensionAction::incrementBlockedResourceCount(ssize_t amount)
+{
+    if (m_blockedResourceCount + amount < 0)
+        return;
```

Comment:
Should this still take the count to zero? What do other browsers do?

Reply:
I think it should be able to take it to zero - that would clear the badge text.

Reply:
I mean, if `m_blockedResourceCount` is 1, and `amount` is -2, should it just reset `m_blockedResourceCount` to 0? Right now it will stay at 1 with the return.

Reply:
I will see what Chrome does.

Reply:
What you described is what Chrome does. I will make that change.